pkg/services/ngalert/eval/eval.go

```diff
@@ -24,28 +25,72 @@ import (
 
 var logger = log.New("ngalert.eval")
 
-//go:generate mockery --name Evaluator --structname FakeEvaluator --inpackage --filename evaluator_mock.go --with-expecter
-type Evaluator interface {
-	// ConditionEval executes conditions and evaluates the result.
-	ConditionEval(ctx EvaluationContext, condition models.Condition) Results
-	// QueriesAndExpressionsEval executes queries and expressions and returns the result.
-	QueriesAndExpressionsEval(ctx EvaluationContext, data []models.AlertQuery) (*backend.QueryDataResponse, error)
+type EvaluatorFactory interface {
```

Comment:
I'm not sure that we want `Validate` in the factory interface because it doesn't create anything? What if we had a separate interface `Validator` for validation and then a function type for creating evaluators like `type NewConditionEvaluatorFunc func(ctx EvaluationContext, condition models.Condition) (ConditionEvaluator, error)`.

Reply:
Good point! There is another interface in api package that declares only Validate method, and I had to add it to this interface to not touch more code. I will address your comment in a follow-up PR